From 0f7481f4384812b5f2a7e6bda211c237fc9ede71 Mon Sep 17 00:00:00 2001
From: Stilyan Uzunov <quic_suzunov@quicinc.com>
Date: Thu, 20 Feb 2025 14:07:34 +0000
Subject: [PATCH] gstreamer1.0-plugins-base: Add meson option to build all
 plugins

- Default value is to build all plugins
- If reduced set of plugins is selected, then only are build:
    gst-libs/gst/video

Upstream-Status: Inappropriate [Qualcomm specific]

Signed-off-by: Stilyan Uzunov <quic_suzunov@quicinc.com>
---
 gst-libs/gst/meson.build | 30 +++++++++++++++++-------------
 gst/meson.build          | 16 ++++++++++------
 meson.build              | 37 +++++++++++++++++++++++--------------
 meson_options.txt        |  4 ++++
 4 files changed, 54 insertions(+), 33 deletions(-)

diff --git a/gst-libs/gst/meson.build b/gst-libs/gst/meson.build
index 42c0a6f..2dd2b7f 100644
--- a/gst-libs/gst/meson.build
+++ b/gst-libs/gst/meson.build
@@ -1,13 +1,17 @@
-subdir('tag')
-subdir('fft')
-subdir('video')
-subdir('audio')
-subdir('rtp')
-subdir('pbutils')
-subdir('sdp')
-subdir('rtsp')
-subdir('riff')
-subdir('app')
-subdir('allocators')
-# FIXME: gl deps are automagic
-subdir('gl')
+if build_all_plugins
+  subdir('tag')
+  subdir('fft')
+  subdir('video')
+  subdir('audio')
+  subdir('rtp')
+  subdir('pbutils')
+  subdir('sdp')
+  subdir('rtsp')
+  subdir('riff')
+  subdir('app')
+  subdir('allocators')
+  # FIXME: gl deps are automagic
+  subdir('gl')
+else
+  subdir('video')
+endif
diff --git a/gst/meson.build b/gst/meson.build
index 5eec601..c3f1cb0 100644
--- a/gst/meson.build
+++ b/gst/meson.build
@@ -1,6 +1,10 @@
-foreach plugin : ['adder', 'app', 'audioconvert', 'audiomixer', 'audiorate', 'audioresample',
-                  'audiotestsrc', 'compositor', 'dsd', 'debugutils', 'encoding', 'gio', 'overlaycomposition',
-                  'pbtypes', 'playback', 'rawparse', 'subparse', 'tcp', 'typefind',
-                  'videoconvertscale', 'videorate', 'videotestsrc', 'volume']
-  subdir(plugin)
-endforeach
+if build_all_plugins
+  foreach plugin : ['adder', 'app', 'audioconvert', 'audiomixer', 'audiorate', 'audioresample',
+                    'audiotestsrc', 'compositor', 'dsd', 'debugutils', 'encoding', 'gio', 'overlaycomposition',
+                    'pbtypes', 'playback', 'rawparse', 'subparse', 'tcp', 'typefind',
+                    'videoconvertscale', 'videorate', 'videotestsrc', 'volume']
+    subdir(plugin)
+  endforeach
+else
+  subdir('videorate')
+endif
diff --git a/meson.build b/meson.build
index 3f6730c..d23196a 100644
--- a/meson.build
+++ b/meson.build
@@ -504,26 +504,35 @@ else
   pathsep = ':'
 endif

+build_all_plugins = get_option('build-all-plugins')
+
 python3 = import('python').find_installation()
 subdir('gst-libs')
-subdir('gst')
-subdir('ext')
-subdir('sys')
-subdir('tools')
-subdir('tests')

-# xgettext is optional (on Windows for instance)
-if find_program('xgettext', required : get_option('nls')).found()
-  core_conf.set('ENABLE_NLS', 1)
-  subdir('po')
+if build_all_plugins
+  subdir('gst')
+  subdir('ext')
+  subdir('sys')
+  subdir('tools')
+  subdir('tests')
+
+  # xgettext is optional (on Windows for instance)
+  if find_program('xgettext', required : get_option('nls')).found()
+    core_conf.set('ENABLE_NLS', 1)
+    subdir('po')
+  endif
+  subdir('docs')
+  subdir('scripts')
+else
+  subdir('gst')
 endif
-subdir('docs')
-subdir('scripts')

 base_libraries = ['allocators', 'app', 'audio', 'fft', 'pbutils', 'riff', 'rtp', 'rtsp', 'sdp', 'tag', 'video']
-if build_gstgl
-  core_conf.set('HAVE_GL', 1)
-  base_libraries += 'gl'
+if build_all_plugins
+  if build_gstgl
+    core_conf.set('HAVE_GL', 1)
+    base_libraries += 'gl'
+  endif
 endif

 pkgconfig_plugins_base_libs_variables = [
diff --git a/meson_options.txt b/meson_options.txt
index 17b5743..29f9372 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -85,6 +85,10 @@ option('package-origin', type : 'string', value : 'Unknown package origin', yiel
        description : 'package origin URL to use in plugins')
 option('doc', type : 'feature', value : 'auto', yield: true,
        description: 'Enable documentation.')
+
+# Build options
+option('build-all-plugins', type : 'boolean', yield : true,
+       description : 'Enable build of all plugins')
 option('glib_debug', type : 'feature', value : 'auto', yield : true, description : 'Enable GLib debug infrastructure (see docs/macros.txt)')
 option('glib_assert', type : 'boolean', value : true, yield : true, description : 'Enable GLib assertion (see docs/macros.txt)',
   deprecated: {'enabled' : 'true', 'disabled' : 'false', 'auto' : 'false'},
--
2.47.3

